src/charm.py

```diff
@@ -144,6 +149,7 @@ def _update_config(self, event=None):
         if not self._container.can_connect():
             # Pebble is not ready yet so no need to update config
             self.unit.status = WaitingStatus("waiting for agent container to start")
+            event.defer()
```

Comment:
Why do we need to defer here? If can_connect is false, then we know we will get another call when pebble is ready.

Reply:
You are right, it's not necessary.